src/drivers/clock/epit/epit.c

```diff
@@ -0,0 +1,100 @@
+/**
+ * @file epit.c
+ * @brief Enhanced.Periodic.Interrupt.Timer driver
+ * @author Puranjay Mohan <puranjay12@gmail.com>
+ * @version 0.1
+ * @date 2020-03-05
+ */
+
+#include <drivers/common/memory.h>
+#include <hal/clock.h>
+#include <hal/reg.h>
+#include <kernel/irq.h>
+#include <kernel/printk.h>
+#include <kernel/time/clock_source.h>
+#include <kernel/time/time_device.h>
+
+#include <framework/mod/options.h>
+#include <embox/unit.h>
+
+#define EPIT_BASE		OPTION_GET(NUMBER, base_addr)
+#define EPIT_IRQ		OPTION_GET(NUMBER, irq_nr)
+#define EPIT_TARGET_HZ	OPTION_GET(NUMBER, freq)
+
+#define EPIT_CR		(EPIT_BASE + 0x00) /* Control Register */
+#define EPIT_SR		(EPIT_BASE + 0x04) /* Status Register */
+#define EPIT_LR		(EPIT_BASE + 0x08) /* Load Register */
+#define EPIT_CMPR		(EPIT_BASE + 0x0C) /* Compare Register */
+#define EPIT_CNR		(EPIT_BASE + 0x10) /* Counter Register */
+
+#define EPIT_PRESCALER	66 /* prescaler value of 66 */
+#define EPIT_CR_PRESCLR	(EPIT_PRESCALER << 4) /* Prescaler value of 66 */
+#define EPIT_CR_EN		(1 << 0) /* Enable Bit */
+#define EPIT_CR_ENMOD		(1 << 1) /* Enable Mode Bit */
+#define EPIT_CR_OCIEN		(1 << 2) /* Output compare Interrupt Enable bit*/
+#define EPIT_CR_RLD		(1 << 3) /* Counter Reload Control bit */
+#define EPIT_SR_CLEAR		(1 << 0) /* Clear the status register*/
+#define EPIT_CR_CLKSRC	(1 << 24) /* Use peripheral clock as input, 66MHz */
+#define EPIT_CR_IOVW		(1 << 17) /* Load Immidiate Override*/
+#define EPIT_LR_VALUE		(66000000)/(EPIT_PRESCALER * EPIT_TARGET_HZ)
```

Comment:
What is `66000000`?

Reply:
btw, it's safer to write `#define EPIT_LR_VALUE	 (66000000 / (EPIT_PRESCALER * EPIT_TARGET_HZ))`

Reply:
@alexkalmuk The 66000000 is the frequency of the peripheral clock i.e. 66 Mhz, it is used in the EPIT timer.

Reply:
I have changed the #define as you specified.
Thanks

Reply:
Can we use `#define EPIT_FREQ 66000000` then?

Reply:
Yes, I will change it.

Reply:
done.